src/Compilers/Core/CodeAnalysisTest/DefaultAnalyzerAssemblyLoaderTests.cs

```diff
@@ -28,642 +37,846 @@ public class AssemblyLoadTestFixtureCollection : ICollectionFixture<AssemblyLoad
         }
 
-        [ConditionalFact(typeof(CoreClrOnly))]
+        [Fact]
         public void AssemblyLoading_DependencyInDifferentDirectory()
         {
-            StringBuilder sb = new StringBuilder();
-            var loader = new DefaultAnalyzerAssemblyLoader();
+            Run(static (DefaultAnalyzerAssemblyLoader loader, AssemblyLoadTestFixture testFixture) =>
+            {
+                using var temp = new TempRoot();
+                StringBuilder sb = new StringBuilder();
 
-            var tempDir = Temp.CreateDirectory();
+                var deltaFile = temp.CreateDirectory().CreateFile("Delta.dll").CopyContentFrom(testFixture.Delta1.Path);
+                var gammaFile = temp.CreateDirectory().CreateFile("Gamma.dll").CopyContentFrom(testFixture.Gamma.Path);
 
-            var deltaFile = tempDir.CreateFile("Delta.dll").CopyContentFrom(_testFixture.Delta1.Path);
-            loader.AddDependencyLocation(deltaFile.Path);
-            loader.AddDependencyLocation(_testFixture.Gamma.Path);
-            Assembly gamma = loader.LoadFromPath(_testFixture.Gamma.Path);
+                loader.AddDependencyLocation(deltaFile.Path);
+                loader.AddDependencyLocation(gammaFile.Path);
+                Assembly gamma = loader.LoadFromPath(gammaFile.Path);
 
-            var b = gamma.CreateInstance("Gamma.G")!;
-            var writeMethod = b.GetType().GetMethod("Write")!;
-            writeMethod.Invoke(b, new object[] { sb, "Test G" });
+                var b = gamma.CreateInstance("Gamma.G")!;
+                var writeMethod = b.GetType().GetMethod("Write")!;
+                writeMethod.Invoke(b, new object[] { sb, "Test G" });
 
-            var actual = sb.ToString();
-            Assert.Equal(@"Delta: Gamma: Test G
+                var actual = sb.ToString();
+                Assert.Equal(@"Delta: Gamma: Test G
 ", actual);
 
 #if NETCOREAPP
-            var alcs = DefaultAnalyzerAssemblyLoader.TestAccessor.GetOrderedLoadContexts(loader);
-            Assert.Equal(1, alcs.Length);
-
-            VerifyAssemblies(
-                alcs[0].Assemblies,
-                ("Delta", "1.0.0.0", deltaFile.Path),
-                ("Gamma", "0.0.0.0", _testFixture.Gamma.Path)
-            );
+
+                var alcs = DefaultAnalyzerAssemblyLoader.TestAccessor.GetOrderedLoadContexts(loader);
+                Assert.Equal(1, alcs.Length);
+
+                VerifyAssemblies(
+                    alcs[0].Assemblies,
+                    ("Delta", "1.0.0.0", deltaFile.Path),
+                    ("Gamma", "0.0.0.0", gammaFile.Path)
+                );
+
 #endif
+            });
         }
 
+        /// <summary>
+        /// Similar to <see cref="AssemblyLoading_DependencyInDifferentDirectory"/> except want to validate
+        /// a dependency in the same directory is preferred over one in a different directory.
```

Comment:
You're certain we didn't already have this test? this sounds very very familiar. Also, it might be good to choose one of the Delta1 variants which has different output at runtime. Otherwise, I think it's difficult to be sure if the dependency in the same directory really was preferred when running on framework.

Although I'd imagine there's also a moment-in-time problem when on framework, if another analyzer loaded something with a matching name, there's no opportunity for this analyzer to "override" it.

Reply:
> Also, it might be good to choose one of the Delta1 variants which has different output at runtime

In this case I wanted to use the same DLL in two places because I wanted version to be identical. Didn't want version greater than coming into play. 

> Otherwise, I think it's difficult to be sure if the dependency in the same directory really was preferred when running on framework.

Actually, now that we have isolated `AppDomain` per run we can basically do the same verification we do for .NET Core. We can check the loaded assemblies. Let me do that.

Reply:
> In this case I wanted to use the same DLL in two places because I wanted version to be identical.

IIRC, the test fixture contains multiple "versions" of Delta with exactly the same assembly version but different behavior. That's what I had in mind. But, if we can verify the paths of the loaded assemblies on framework, it's not essential to do this.

Reply:
Yes indeed Delta2B and Delta2 have the same version but different content on a different path. The newest version of the change though allows us to verify paths on both .NET Framework and .NET Core. The .NET Framework code path is less ideal since we don't have the isolation of `AssemblyLoadContext` but I found a way to tease out these values.

Reply:
Actually I found a method to detect if an `Assembly` is in the _LoadFrom_ context vs. _Load_ in .NET Framework so we can be very direct with our tests here now.